components/merkledb/src/db.rs

```diff
@@ -567,12 +521,27 @@ impl Snapshot for FlushedFork {
 }
 
 impl Fork {
+    /// Creates a fork based on the provided `patch` and `snapshot`.
+    ///
+    /// Note: using created fork to modify data already present in `patch` may lead
+    /// to an inconsistent database state. Hence, this method is useful only if you
+    /// are sure that the fork and `patch` interacted with different indices.
+    pub fn from_patch(patch: Patch, snapshot: Box<dyn Snapshot>) -> Self {
```

Comment:
Do we really need `snapshot` parameter here ? We have it in Patch ? 
```rust
Self {
             patch: Patch {
                 changes: patch.changes,
                 snapshot: patch.snapshot,
             },
             working_patch: WorkingPatch::new(),
         }
}
```

Reply:
We can't use snapshot from patch, because it's doesn't have changes from mempool made by `handle_tx`. We can use it only after we will rewrite mempool.

Reply:
Yeah. I got it. It's a really pain. It makes public api ugly.

Reply:
This method is full of dangerous dark corners and looks very ugly. It should not be public method.
And moreover, we have to find solution that does not need such methods.
For example, we can keep fork in transactions pool and at the moment of commit block use this fork as current blockchain state. But during the proposal execution we can rollback changes from fork after that propose will be created.